test/mongodb.test.js

```diff
@@ -265,6 +265,29 @@ describe('mongodb connector', function() {
         ds.disconnect(done);
       });
     });
+
+    it('should prioritize to the database given in the url property', function(done) {
+      var cfg = JSON.parse(JSON.stringify(config));
+      var testDb = 'lb-ds-overriden-test-1';
+      cfg.url = 'mongodb://' + cfg.hostname + ':' + cfg.port + '/' + testDb;
+      var ds = getDataSource(cfg);
+      ds.once('connected', function() {
+        var db = ds.connector.db;
+        var validationError = null;
+        try {
+          db.should.have.property('databaseName', testDb); // check the db name in the db instance
```

Comment:
I suggest we check that `self.settings.database` is also set here, so that we know the database name from the url string takes precedence.

Reply:
`self.settings` is a reflection of the datasource options.
So `self.settings.database` would contain the value given in the original options. The property `self.settings.url` would be the one the has the required value. The whole point is for the Mongo connector to prioritize usage of `url` property in order to connect, against the `database` and not override the settings provided. At least that's what I think

Reply:
yeah IIUC we are on the same page. I was just wondering if we can assert that `self.settings.database` is also set, so that we know `self.settings.url` got prioritized. To be honest, though, this assertion is enough. Any reason why we have to call `ds.ping`?

Reply:
It was just to verify that db connection was open correctly, if you want I can remove it

Reply:
no worries, I'm fine with it. Thanks for clarifying.